src/app/view/map/map-devices.component.ts

```diff
@@ -0,0 +1,114 @@
+import { Component, OnInit } from '@angular/core';
+import * as L from 'leaflet';
+import { DeviceService } from '../../auth/services/device.service';
+import { Device } from '../../models/device.model';
+import { MapDevicesService } from '../../auth/services/map-devices.service';
+import { OrganizationType } from '../../utils/enums/organization-types.enum';
+
+@Component({
+  selector: 'app-map-devices',
+  templateUrl: './map-devices.component.html',
+  styleUrls: ['./map-devices.component.scss'],
+})
+export class MapDevicesComponent implements OnInit {
+  // Map options
+  options: L.MapOptions = {
+    layers: [],
+    zoom: 3,
+    center: L.latLng(20, 0), // Center of the world map
+  };
+
+  // Map instance
```

Comment:
the map component will be used across different part of the frontend. it should have a component which doesn't anything to do with devices or fetching data. it will just receive coordinates and render them on the map

Reply:
it will just render locations of devices that are displayed in the device table